### DISCUSSION TITLE
innie vs outie punctuation in British and American quotes (e.g., equivalent of csquotes)

### DISCUSSION THREAD

**Discussion by reagle** [jgm/pandoc#9600] [Q&A]
Has there been discussion or a filter that allows the author to write in American (punctuation tucked-in) and British/Wikipedia (punctuation outside) styles, and switch the output to the other? (Natively, and working in all output formats?) This would be the equivalent of [csquotes](https://ctan.org/pkg/csquotes?lang=en).

**Comment by denismaier** [comment]
I've tried something like that a couple of years ago:
https://github.com/denismaier/pandoc-lua-move-punctuation-after-citations

Gesendet von Outlook für Android<https://aka.ms/AAb9ysg>
________________________________
From: Joseph Reagle ***@***.***>
Sent: Friday, March 22, 2024 5:33:39 PM
To: jgm/pandoc ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [jgm/pandoc] innie vs outie punctuation in British and American quotes (e.g., equivalent of csquotes) (Discussion #9600)


Has there been discussion or a filter that allows the author to write in American (punctuation tucked-in) and British/Wikipedia (punctuation outside) styles, and switch the output to the other? (Natively, and working in all output formats?) This would be the equivalent of csquotes<https://ctan.org/pkg/csquotes?lang=en>.

**Comment by reagle** [comment]
@denismaier, thank you. I think pandoc does something like that (around citations) natively now...? I'm asking about placement even when there is no punctuation.

**Comment by denismaier** [comment]
Yes, pandoc does that natively around citations and also with regards to footnotes more generally. It has already done so for quite some time now. But, my filter works (or used to work, haven't used it recently) with opposite assumptions. (See [here](https://github.com/jgm/pandoc-citeproc/issues/354) for the reasoning.)

How would you design the interface of such a more general filter? Maybe something like this?

```
[This is a full quotation.]{.csquote}

[This is a another full quotation]{.csquote punct="."}

Here comes a [partial quotation]{.csquote}.

Here comes another [partial quotation]{.csquote tpunct="."}
```

**Reply by reagle** [reply]
Honestly, I'd rarely use it if I had to use such cumbersome syntax --- so not "equivalent" to csquotes I guess. I wonder if, instead, since pandoc is excellent at turning straight quotes could terminating punctuation (with citation and note processing) be configurable by a single language option. (Probably not, and if so, easier said than done! 🙂 )

**Reply by denismaier** [reply]
You can probably do that. The question is just this: how should pandoc know whether terminal punctuation belongs inside or outside? It can probably work in simple cases, but I fear heuristics will only get you so far ...

**Reply by reagle** [reply]
A command line and metadata option? You're right about simple will only get you so far. cslquote has probably a hundred options, and that's not what I'd want.

**Comment by dstark** [comment]
Perhaps related to https://github.com/jgm/pandoc/issues/7005?
